src/Collapse.js

```diff
@@ -34,7 +35,12 @@ const Collapse = React.createClass({
 
 
   componentWillReceiveProps({isOpened}) {
-    this.setState({isOpenedChanged: isOpened !== this.props.isOpened});
+    var isOpenedChanged = isOpened !== this.props.isOpened;
+    this.setState({isOpenedChanged}, () => {
+      if (this.props.onHeightReady && isOpenedChanged) {
+        this.props.onHeightReady(isOpened ? this.state.height : 0);
```

Comment:
I think doing "side-effect" (notifying parent) in componentWillReceiveProps is dangerous. It also would potentially end up in recursion.

Potentially this could be added to some other lifecycle method.